pkg/backup/backupdest/backup_destination.go

```diff
@@ -224,8 +233,39 @@ func ResolveDest(
 	}
 	prevBackupURIs = append([]string{plannedBackupDefaultURI}, prevBackupURIs...)
 
+	baseEncryptionOptions, err := backupencryption.GetEncryptionFromBase(
+		ctx, user, execCfg.DistSQLSrv.ExternalStorageFromURI, prevBackupURIs[0],
+		encryption, kmsEnv,
+	)
+	if err != nil {
+		return ResolvedDestination{}, err
+	}
+
+	if startTime.IsEmpty() {
+		// TODO (kev-cao): Once we have completed the backup directory index work, we
+		// can remove the need to read an entire backup manifest just to fetch the
+		// start time. We can instead read the metadata protobuf.
+		mem := execCfg.RootMemoryMonitor.MakeBoundAccount()
+		defer mem.Close(ctx)
+		precedingBackupManifest, size, err := backupinfo.ReadBackupManifestFromURI(
+			ctx, &mem, prevBackupURIs[len(prevBackupURIs)-1], user,
```

Comment:
could you add one note: that we're assuming that FindPriorBackups returns prev backups in ascending end time order, i.e. the backup with the highest end time will be last in the list. We make this assumption later in the codebase as well.

Reply:
Uh oh:
>The order that results are passed to the callback is undefined.

https://github.com/kev-cao/cockroach/blob/backup/incremental-path-suffix/pkg/cloud/external_storage.go#L81

Reply:
Oh, lord have mercy.  We're good. Thought we had bad bad bug for a second.

https://github.com/kev-cao/cockroach/blob/backup/incremental-path-suffix/pkg/backup/backupdest/incrementals.go#L113

Reply:
Maybe add an extra line to FindPriorBackups explaining the order in which backups are returned lol